controllers/users.ts

```diff
@@ -0,0 +1,82 @@
+import { NextFunction, Request, Response } from 'express';
+import httpStatus from 'http-status';
+import { isValidObjectId } from 'mongoose';
+import { userModel } from '../models/users';
+
+export const createUser = async (request: Request, response: Response, next: NextFunction) => {
+	const data = request.body;
+
+	try {
+		const newUser = await userModel.create(data);
+		response.status(httpStatus.CREATED).send({ ...newUser, password: undefined });
+	} catch (error) {
+		next(error);
+	}
+};
+
+export const getUsers = async (
+	request: Request<{}, {}, {}, { userId?: string; username?: string; email?: string; password: never }>,
```

Comment:
use ur newly created type

Reply:
it would look like `request: Request<{}, {}, {}, Partial<UserDetails & { userId: string }>>`

`UserDetails` is a type I added that contains only `email` and `password`, and since the `User` type itself contains `_id` I need to add it like this.

I'm thinking of a few solutions:
1. keep it as it was
2. when querying a user by its id, it will look like `/users?_id=234134...`
3. split this controller into two, having one only find by Id `/users/:id` and one by username / email

Reply:
3 sounds legit. Also about ur type, I see u resolved my type comment, so u can just use it and `Pick` these fields for UserDetails